add specification docs, spline layer spec and functions for spline io

alisterburt opened the pull request:
Hey @kevinyamauchi - I think this is good to go, let me know what you think of the outline for the spec in the docs and if you think anything is missing

Once this is in I will implement a single io plugin entry point in a separate plugin rather than implementing one for every annotator

kevinyamauchi reviewed (commented):
Thanks, @alisterburt ! I think this looks good! A couple of comments below, but I think this is basically good to go

- I think you missed the `zarr` dependency in the `setup.cfg` (I think that's the cause of the test failure)
- I think we should have an `n3d_metadata` (or some other name) dict in the layer metada that holds all of the n3d-related metadata rather than storing everything on the root level of `layer.metadata`. What do you think?

alisterburt commented:
Thanks for the suggestions! just FYI this is not yet ready for review, didn't have time to completely finish updating so it's still in limbo

alisterburt commented:
@kevinyamauchi all looking good to go here, feel free to take another look or merge as is! I implemented your suggestions and everything is now stored either in the layer features or an n3d_metadata dict in the layer metadata 

I removed all magic strings and factored them out as constants either on the model or in the annotator io module
